### PR TITLE
fix: [PIE-4470]: fixed NPE

### PR DESCRIPTION
**Pull request by shalini-agr**
NotificationChannelWrapper.getType() returns null when notification method is given as null {} in pipeline yaml.
Fixed this by putting a null check on wrapper.getType().

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/35638)

### CODE REVIEW

**Comment by shalini-agr** [comment]
trigger messagecheck

File: 800-pipeline-service/src/main/java/io/harness/pms/notification/NotificationHelper.java [hunk 1]
```diff
@@ -116,15 +116,19 @@ private void sendNotificationInternal(List<NotificationRules> notificationRulesL
...
+              accountIdentifier, orgIdentifier, projectIdentifier, templateId, notificationContent, ambiance);
+          log.info(
+              "Sending notification via notification-client for plan execution id: {} ", ambiance.getPlanExecutionId());
+          try (AutoLogContext ignore = AmbianceUtils.autoLogContext(ambiance)) {
+            notificationClient.sendNotificationAsync(channel);
+          } catch (Exception ex) {
+            log.error("Unable to send notification because of following exception", ex);
+          }
+        } else {
+          log.error("Unable to send notification because notification type is null");
```
Comment (namanvermah) [inline 800-pipeline-service/src/main/java/io/harness/pms/notification/NotificationHelper.java:131]: If there's any other info here, like pipeline event type, pipeline identifier, put that here in the error log. Will making debugging easier.

**Comment by shalini-agr** [comment]
*[`800-pipeline-service/src/main/java/io/harness/pms/notification/NotificationHelper.java` line 131 at r1](https://reviewable.io/reviews/harness/harness-core/35638#-N7_D0fdMV7tabXvHTMW-r1-131:-N7_SFFi0JF9qf8WNUFa:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/31d81a4476dcfb1c20046c810a6c31d7ba209ca4/800-pipeline-service/src/main/java/io/harness/pms/notification/NotificationHelper.java#L131)):*
<details><summary><i>Previously, namanvermah (Naman Verma) wrote…</i></summary><blockquote>

If there's any other info here, like pipeline event type, pipeline identifier, put that here in the error log. Will making debugging easier.
</blockquote></details>

Done.

**Review by brijeshdhakar — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35638)* status: 1 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by namanvermah — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35638#-:-N7nVwwT9DYQyaiRn7rj:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**